src/arg-types/utils.ts

```diff
@@ -0,0 +1,78 @@
+import type {ArgTypes} from "@storybook/csf"
+import {PropItem} from "react-docgen-typescript";
+
+export type FilterPredicate = (prop: PropItem) => boolean
+
+const validAttributes = (prop: PropItem) => {
+    if (prop.parent) {
+        return ['ButtonHTMLAttributes', 'HTMLAttributes', 'AriaAttributes'].includes(prop.parent.name)
+    }
+    return true
+}
+
+export const propsToArgTypes = (props: Record<string, PropItem>, filter?: FilterPredicate): ArgTypes => {
+    const filterFn = filter ?? validAttributes
+    const entries = Object.entries(props);
+    return entries
+        .reduce((result, current) => {
+            const [propName, prop] = current
+
+            // Filter out props
+            if (!filterFn(prop)) {
+                return result
+            }
+
+            const control = mapControlForType(prop)
+            result[propName] = {...prop, ...control}
+            return result
+        }, {} as ArgTypes);
+}
+
+const matchers = {
```

Comment:
needs to be extended to support all controls?

Reply:
That's how it's done in Storybook now. I would not do it here at all TBH.

Reply:
hmm, does this compromise our ability to render controls in both correctly? I guess if we do a better job at rendering than storybook its not a problem, unless we become incompatible